src/components/Icon/Icon.tsx

```diff
@@ -39,17 +41,107 @@ export const Icon: (props: IIconProps) => JSX.Element = (props: IIconProps) => {
         iconWidth = props.width || '16px';
     }
 
+    const numberString = props.notificationNumber === undefined ? '' : props.notificationNumber < 10 ?
+                            props.notificationNumber.toLocaleString() : '9+';
+    const defaultStyleObject = DefaultIconNumberStyleObject(props.iconSize, numberString.length);
+    let styleObject: NotificationBubbleStyleObject = {numberStyleObject: {}, containerStyleObject: {}, bubbleStyleObject: {}};
+
+    if (props.notificationBubbleStyleObject) {
```

Comment:
maybe use Object.assign or something to avoid manual setting of the values?

Reply:
the point here is the merge option. object.assign will override any default option, while this allows for partial style object, if i'm not wrong

Reply:
yes, the point is that you can assign only, for example, color to blue and that all other values are default... that's why I need to assign manualy

Reply:
i'm afraid i do not understand.
is this not the same?

```javascript
 styleObject: NotificationBubbleStyleObject = {
containerStyleObject: {
    ...defaultStyleObject.containerStyleObject
}}

if(props.notificationBubbleStyleObject.containerStyleObject) {
 object.assign(styleObject.containerStyleObject, props.notificationBubbleStyleObject.containerStyleObject)
}
;```

Reply:
if props.notificationBubbleStyleObject.containerStyleObject has only width set, then when it is asigned to styleObject.containerStyleObject, styleObject.containerStyleObject will also have only width set. what we wanted here is to have styleObject.containerStyleObject with default values for height, position, display and alignItems and only width from props.notificationBubbleStyleObject.containerStyleObject. this way when you wish to set only width, you can, you don't have to set height, position, display and alignItems too.

Reply:
I dont get it.
that is why i used the spread operator when initialiing containerStyleObject, to set the default values. Object.assign will the use the partial object to override the rest of them.

Object.assign({width:10, height:10}, {widht:12}) will produce {widht:12, height: 10}

true, Object.assign({width:10, height:10}, {widht:null}) will produce {widht:null, height: 10}, but i don't see this as a problem.

Reply:
unless you specifically wanted to support this case.

Reply:
you are right. I apologize, I didn't realize it right away. I will change it. Thank you.